src/tests/test_des.cpp

```diff
@@ -0,0 +1,45 @@
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/nar/des/des.h"
+#include "all_csv_configs.h"
+#include "config/names.h"
+
+namespace tests {
+
+class DESTest : public ::testing::Test {
+protected:
+    static algos::StdParamsMap GetParamMap(CSVConfig const& csv_config, double minsup,
+                                           double minconf, unsigned int popSize,
+                                           unsigned int evalNum, double crossProb, double diffScale,
+                                           algos::des::DifferentialStrategy diffStrategy) {
+        using namespace config::names;
+        return {{kCsvConfig, csv_config},          {kMinimumSupport, minsup},
+                {kMinimumConfidence, minconf},     {kPopulationSize, popSize},
+                {kMaxFitnessEvaluations, evalNum}, {kCrossoverProbability, crossProb},
+                {kDifferentialScale, diffScale},   {kDifferentialStrategy, diffStrategy}};
+    }
+
+    template <typename... Args>
+    static std::unique_ptr<algos::des::DES> CreateAlgorithmInstance(Args&&... args) {
+        return algos::CreateAndLoadAlgorithm<algos::des::DES>(
+                GetParamMap(std::forward<Args>(args)...));
+    }
+};
+
+TEST_F(DESTest, LaunchTest) {
+    auto algorithm = CreateAlgorithmInstance(kAbalone, 0.5, 0.6, 100u, 100u, 0.9, 0.5,
+                                             algos::des::DifferentialStrategy::rand1Bin);
+    algorithm->Execute();
+    std::vector<std::string> result;
+    for (auto i : algorithm->GetNARVector()) {
+        result.push_back(std::to_string(i.GetQualities().fitness));
+    }
```

Comment:
I would add a method to `DESTest`, seems like it can be reused later for other tests:
```cpp
    static std::vector<std::string> ExtractFitnessValues(
            std::vector<algos::des::NAR> const& nar_vector) {
        std::vector<std::string> fitness_values;
        fitness_values.reserve(nar_vector.size());
        for (auto const& nar : nar_vector) {
            fitness_values.push_back(std::to_string(nar.GetQualities().fitness));
        }
        return fitness_values;
    }
```
```suggestion
    auto fitness_values = ExtractFitnessValues(algorithm->GetNARVector());
```